lib/ecl/ecl_util.cpp

```diff
@@ -373,10 +361,7 @@ static bool valid_base(const char * input_base, bool * upper_case) {
 */
 
 static char * ecl_util_alloc_filename_static(const char * path, const char * base , ecl_file_enum file_type , bool fmt_file, int report_nr, bool must_exist) {
-  bool upper_case;
-  if (!valid_base(base, &upper_case))
-    return NULL;
-
+  bool upper_case = base_has_upper(base);
```

Comment:
I suggest that you move this call to the place where `upper_case` is used. Also, the standard is `ALL_CAPS_NAMES`, so we should really check if `AlL_CaPs_NameS` should yield `AlL_CaPs_NameS.DATA`, `AlL_CaPs_NameS.DaTa` or `AlL_CaPs_NameS.data`. I would expect the first option.

@magnesj Could you comment on this?

Reply:
I think that filenames on disk can come from a vast range of sources, and import of these files should be as flexible as possible.

Reply:
The problem is that it is an established assumption in _libecl_ that the extension casing can be deduced from the base casing. Hence, there is a huge difference in effort if we are to support the base `MixedCase` and we can assume that when mixed case the datafile extension is either always i.e. `DATA` or always i.e. `data`. For example:
 - `ALL_UPPER` -> `ALL_UPPER.DATA`,
 - `all_lower` -> `all_lower.data` and
 - `SOME_lower` -> `SOME_lower.DATA`.

Of course, we can also change it so that `SOME_lower` always assumes `SOME_lower.data`. The problem with allowing both is that then information regarding extensions have to be persisted and propagated through the code. In addition, the consumer of the API would have to specify when writing the summary data what extension they want.

I suggest we finish of this PR with the continued assumption that extension casing is entirely deduced by base name and that this is validated. You still get full flexibility regarding the base name.

Then we can discuss if you want support beyond this afterwards as this will require quite a bit more... Does this sound good?

Reply:
Thanks for the explanation. This PR is an improvement, and I think this will fix the import issues for many users. Thanks.

Reply:
No problem at all! Sorry for the staleness of this PR. I'll make sure the task is well defined and follow it up so that we get it in soon :+1: